homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -26,20 +35,29 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-WEATHER_UPDATE_INTERVAL = timedelta(minutes=10)
-
 
 class WeatherUpdateCoordinator(DataUpdateCoordinator):
     """Weather data update coordinator."""
 
-    def __init__(self, owm, latitude, longitude, hass):
+    def __init__(self, owm, latitude, longitude, forecast_mode, hass):
         """Initialize coordinator."""
         self._owm_client = owm
         self._latitude = latitude
         self._longitude = longitude
+        self._forecast_mode = forecast_mode
+        if forecast_mode == "freedaily":
+            # Auto convert to One Call for freedaily users
+            self._forecast_mode = "onecall_daily"
+        self._forecast_limit = None
+        if forecast_mode == "daily":
+            self._forecast_limit = 15
+
+        self._weather_update_interval = timedelta(minutes=5)
+        if self._forecast_mode == "freedaily":
```

Comment:
This if cannot be true, as you change in line 50 the value to _onecall_daily_

Reply:
Good point. I'll update it.